pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -105,6 +108,38 @@ public ResultSetGroup execute(Request request)
     return execute(null, request);
   }
 
+  /**
+   * Extract tableName from the given query
+   * @param query
+   * @return tableName
+   * @throws PinotClientException 
+   */
+  public String getTableNameFromQuery(String query) {
+    PinotQuery pinotQuery;
+
+    try {
+      pinotQuery = CalciteSqlParser.compileToPinotQuery(query);
+    } catch (Exception e) {
+      LOGGER.info("Caught exception while compiling SQL request: {}, {}", query, e.getMessage());
+      throw new PinotClientException("Caught exception while compiling SQL request");
+    }
+
+    DataSource dataSource;
+    
+    try {
+      dataSource = pinotQuery.getDataSource();
```

Comment:
When this will throw an exception?

Reply:
Thanks for pointing this out. It was actually not throwing the exception.